lib/createExplorer.js

```diff
@@ -47,9 +47,6 @@ module.exports = function (options) {
         const directory = (searchPathIsDirectory)
           ? absoluteSearchPath
           : path.dirname(absoluteSearchPath);
-        if (directoryCache && directoryCache.has(directory)) {
-          return directoryCache.get(directory);
-        }
```

Comment:
We cannot remove these lines. Using the directory cache is essential to prevent lots and lots of unnecessary file system activity.

Why is the cache causing this problem? What is the actual problem? And can you reproduce it in a test?